### PR TITLE
Add tools for maintaining the docs

### PR DESCRIPTION
**Pull request by ST-DDT**
Currently everyone maintains a private copy of tools that he/she needs for maintenance tasks.

This is bad in multiple ways:

* Everyone has to write it himself
* Improvements don't get shared
* Team-members tend to change from time to time
* It cannot be used in automation

Thats why I started this attempt to publish the required sources in the repo as well.

------------------------------------------------------------

## Existing tools

* [x] Javadoc Checker - #640 #666 
* [x] Compile Code Checker - #695
* [x] ~`sponge-conf.rst` - Definition List Generator - #718 #732~ (different PR)

~@NariKiro It would be nice if you could upload your sources for the Definition List Generator somewhere/here (once the remaining questions have been resolved)~

------------------------------------------------------------

## Remaining questions

@SpongePowered/docs 

* [x] Store it in the Docs repo itself or in a separate repo/repo per tool?
  * In the repo
* [x] Which coding conventions should be used?
  * Sponge's but less strict
* [x] Restrictions on used applications such as jdk or build tools?
  * None
* [x] Which license should be used?
  * Any free license, preferable MIT or unlicense
* [x] ~Did I forget something?~

### CODE REVIEW

**Comment by Spongy** [comment]
A preview for this pull request is available at https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/9f1778c/.

Here are some links to the pages that were modified:

- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/9f1778c/contributing/implementation/datamanipulator.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/9f1778c/plugin/commands/creating.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/9f1778c/plugin/data/custom/serialization.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/9f1778c/plugin/entities/ai.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/9f1778c/plugin/injection.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/9f1778c/plugin/items/creating.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/9f1778c/plugin/ray-tracing.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/9f1778c/plugin/services.html

_Since the preview frequently changes, please link to [this comment](https://github.com/SpongePowered/SpongeDocs/pull/754#issuecomment-416051792), not to the direct url to the preview._

File: tools/javadoc-checker/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+Javadoc-Checker
+===============
+
+A helper to check whether the docs contain only valid javadoc references.
```
Comment (Inscrutable) [inline tools/javadoc-checker/README.md]: Isn't the function really to *find invalid javadoc references*?

File: tools/javadoc-checker/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+-----
+
+Either use a precompiled jar:
+
+````bash
+mvn clean package
+java -Dlog4j.skipJansi=false -jar target/javadock-checker....jar [path to SpongeDocs/source]
+````
+
+or run it directly from the command lines:
```
Comment (Inscrutable) [inline tools/javadoc-checker/README.md]: lines -> line

**Review by Inscrutable — COMMENTED** [review]
Now we have the sharpest tools on the block. 🛠  Someone still needs to review the code content.

**Comment by Inscrutable** [comment]
I am inclined to keep the tools under the docs repository unless that causes technical difficulties. We don't want to scatter stuff too widely. Licensing is up to authors, but we'd prefer MIT if possible (or unlicense).

File: tools/javadoc-checker/pom.xml [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+		<dependency>
+			<groupId>org.apache.logging.log4j</groupId>
+			<artifactId>log4j-slf4j-impl</artifactId>
+			<version>2.11.1</version>
+		</dependency>
+		<dependency>
+			<groupId>org.fusesource.jansi</groupId>
+			<artifactId>jansi</artifactId>
+			<version>1.17.1</version>
```
Comment (ST-DDT) [inline tools/javadoc-checker/pom.xml]: Maybe this can be removed.

File: tools/javadoc-checker/src/main/resources/log4j2.xml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration status="INFO">
+	<Appenders>
+		<Console name="Console" target="SYSTEM_OUT">
+			<PatternLayout
+				pattern="[%level@%c{1}] %highlight{%msg%n%throwable}" />
```
Comment (ST-DDT) [inline tools/javadoc-checker/src/main/resources/log4j2.xml]: Try adding `disableAnsi="false"`

**Comment by ST-DDT** [comment]
Compile Code Checker @ 50%

**Comment by ST-DDT** [comment]
Compile Code Checker @ 100%

---------------------------

@NariKiro Do you want to append your tool/code here or do you want to push it in a separate PR?

**Comment by 12awsomeman34** [comment]
I will PR when I find the opportunity.

**Comment by ST-DDT** [comment]
Then we only need a review for the actual code.
Do we need unit tests for those tools?

**Comment by ST-DDT** [comment]
Updated for API 7.1

**Comment by ST-DDT** [comment]
Updated to fix merge conflicts.
Some last change requests?

**Pull request merged**